### DISCUSSION TITLE
[Help]: Accessibility tabbing broken when slides contain a button or link

### DISCUSSION THREAD

**Discussion by RATIU5** [davidjerleke/embla-carousel#799] [Help] [answered]
### Summary

I am trying to make an accessible carousel slider with Embla Carousel in React. I'm running into an issue though.
I have a link inside each slide. When I start tabbing through all tabbed elements on the page, I currently go through this order:
**Left Arrow -> Slide 1 Link -> Slide 2 Link -> Slide 3 Link -> Right Arrow**

I instead want this:
**Left Arrow -> Slide 1 Link -> Right Arrow**

I believe this is because the slides are inline inside the container and don't have `display: none;` when not visible.
Is there a way to make them more accessible without having to write a custom carousel, or needing to rewrite a large portion of the library?

I have tried to pass in an Aria-hidden attribute to the links depending if the current slide is visible or not, but that did not work.

### If applicable, which variants of Embla Carousel are relevant to this question?

- [X] embla-carousel **(Core)**
- [X] embla-carousel-react
- [ ] embla-carousel-vue
- [ ] embla-carousel-svelte
- [ ] embla-carousel-autoplay
- [ ] embla-carousel-auto-scroll
- [ ] embla-carousel-solid
- [ ] embla-carousel-auto-height
- [ ] embla-carousel-class-names
- [ ] embla-carousel-docs **(Documentation)**
- [ ] embla-carousel-docs **(Generator)**

**Comment by sarussss** [comment, marked as answer]
1. Set tabindex = 0
`<div class="embla" tabindex="0">`

2. Custom js and css, you can try
```
.embla__container:not(.disable-hidden) .embla__slide[aria-hidden='true'] {
  visibility: hidden;
  opacity: 0;
}
```
[Demo](https://stackblitz.com/edit/sarusss-autoplay-vk372g?file=main.js)

Hope it's useful to you.

**Reply by RATIU5** [reply]
Thanks! I was able to work a similar solution to this. Help much appreciated.

**Comment by davidjerleke** [comment]
@RATIU5 I'm not sure what you expect? Could you elaborate? Because here's a native CSS scroll snap carousel [example](https://2t4lwt.csb.app/) and this is how the native browser also behaves. This is the order of the tab focus elements:

**Previous button -> Slide 1 Link -> Slide 2 Link -> Slide 3 Link -> Next button**

**Reply by RATIU5** [reply]
My goal was to show one slide link instead of all before the next button. I have read several articles on what's best for accessibility and what I concluded was something like this:
**Previous Button -> Slide 1 Link -> Next Button (pressed & shift-tab) -> Slide 2 Link**
